src/Maker/MakeTest.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Maker;
+
+use Symfony\Bundle\FrameworkBundle\Test\WebTestAssertionsTrait;
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Exception\RuntimeCommandException;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\InputAwareMakerInterface;
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Component\BrowserKit\History;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\CssSelector\CssSelectorConverter;
+use Symfony\Component\Panther\PantherTestCaseTrait;
+
+/**
+ * MakeTest.
+ *
+ * @author Kévin Dunglas <kevin@dunglas.fr>
+ * @author Javier Eguiluz <javier.eguiluz@gmail.com>
+ * @author Ryan Weaver <weaverryan@gmail.com>
+ */
+final class MakeTest extends AbstractMaker implements InputAwareMakerInterface
+{
+    public static function getCommandName(): string
+    {
+        return 'make:test';
+    }
+
+    // TODO: remove this method when removing make:unit-test and make:functional-test
+    public static function getCommandAliases(): iterable
+    {
+        yield 'make:unit-test';
+        yield 'make:functional-test';
+    }
+
+    public static function getCommandDescription(): string
+    {
+        return 'Creates a new test class';
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConf)
+    {
+        $command
+            ->addArgument('name', InputArgument::OPTIONAL, 'The name of the test class (e.g. <fg=yellow>BlogPostTest</>)')
+            ->addArgument('type', InputArgument::OPTIONAL, 'The type of test: <fg=yellow>unit</> (PHPUnit), <fg=yellow>integration</> (WebTestCase) or <fg=yellow>e2e</> (PantherTestCase)')
```

Comment:
I think we need to have a clear terminology throughout the official Symfony resources. In https://github.com/symfony/symfony-docs/pull/14731 and during discussions with @kbond, @Nyholm and @weaverryan, we came up with these terms:

* *Unit tests*, the PHPUnit tests we all know
* *Integration tests*, either PHPUnit tests without too much mocking or `KernelTestCase`. These tests test a combination of multiple classes
* *Application tests* (or "functional tests"), uses `WebTestCase` or `PantherTestCase`. These tests test a whole application, from booting Symfony to sending the response. Whether they use javascript or not is mostly an "implementation detail" of that test.

This slightly differs from the terms used here (integration is the non-javascript "application" variant, and e2e and the javascript "application" variant). I would love to hear your views on this subject (we need to have good terminology here, so it's open for all discussion imho).

Reply:
I commented about that in the Google Docs this morning. I do agree that we need to use the same terminology everywhere. However, I don't like that proposal very much.

1. we'll introduce new confusing terms while they are terms already broadly used and well-defined. I used unit/integration/e2e as in https://martinfowler.com/articles/practical-test-pyramid.html. Using unit/service/UI as in Mike Cohn's original pyramid would work too. But reinventing new terms add more confusion than it solves.
2. we need to make a clear difference between PHP-based integrations tests (`KernelTestCase` / `WebTestCase` / `ApiTestCase`) and UI and e2e tests (Panther). They don't serve the same purpose, and even if they can share the same API, with E2E test you also need specific helpers to interact with the JavaScript application: you need to be able to test that clicking on a button triggers an AJAX request that results the sending of a message...

Maybe this PR misses some test cases, as far as I know the following test cases exist:

* `TestCase`: basic PHPUnit test
* `KernelTestCase`:  to use Symfony services
* `WebTestCase`: to run browser-like scenarios, but don't execute client-side code
* `ApiTestCase`: to run API-oriented scenarios (needs API Platform)
* `PantherTestCase`:  to run e2e scenarios, using real-browsers or HTTP clients, and a real web server

Instead of unit/integration/e2e or unit/service/ui, maybe this generator could use allow to select the `TestCase` to use as base class? WDYT?

Reply:
I think a lot of users will question what `e2e` is. They may get its end to end, but not know exactly what that is or how it differs from a functional, integration, and/or acceptance test. I don't think we need to reinvent the wheel - but defiantly provide a brief statement of what that is + the usual "how to" in symfony docs.

Reply:
I am also not very familiar with e2e - i'm familiar with "functional".

But, my main point is that this is a small detail: most people will not pass an argument. They will, instead, allow the maker to ask interactively what type they want. And when we do THAT, we can allow all the names (though, we will need to choose a "main" name.